db/wal_manager.cc

```diff
@@ -418,6 +418,47 @@ Status WalManager::ReadFirstRecord(const WalFileType type,
   return s;
 }
 
+Status WalManager::GetWalFile(uint64_t number, LogFile** log_file) {
+  if(!number) {
+    return Status::PathNotFound("log file not available");
+  }
+
+  Status s;
+  *log_file = nullptr;
+
+  uint64_t size_bytes;
+  WalFileType log_type = kAliveLogFile;
+  s = env_->GetFileSize(LogFileName(db_options_.wal_dir, number), &size_bytes);
```

Comment:
Do we want to get file size from the file system? Can we, for example, remember it in LogWriter?